Bottom: 43cb54643a5ff6b56c3db029303a251176b11428
Top:    9211707585a0c931aac7160ab30203227c55130e
Author: Jason Wang <jasowang@redhat.com>
Date:   2016-06-29 14:48:25 +0800

Refresh of tun-introduce-tx-skb-ring

---

diff --git a/include/linux/ptr_ring.h b/include/linux/ptr_ring.h
index f156bec..afb2519 100644
--- a/include/linux/ptr_ring.h
+++ b/include/linux/ptr_ring.h
@@ -409,13 +409,12 @@ static inline int ptr_ring_resize_multiple(struct ptr_ring **rings, int nrings,
 			goto nomem;
 	}
 
-	spin_lock_irqsave(&(rings[i])->producer_lock, flags);
-
-	for (i = 0; i < nrings; ++i)
+	for (i = 0; i < nrings; ++i) {
+		spin_lock_irqsave(&(rings[i])->producer_lock, flags);
 		queues[i] = __ptr_ring_swap_queue(rings[i], queues[i],
 						  size, gfp, destroy);
-
-	spin_unlock_irqrestore(&(rings[i])->producer_lock, flags);
+		spin_unlock_irqrestore(&(rings[i])->producer_lock, flags);
+	}
 
 	for (i = 0; i < nrings; ++i)
 		kfree(queues[i]);
